sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -339,8 +339,24 @@ class AstBuilder extends SqlBaseBaseVisitor[AnyRef] with Logging {
       case SqlBaseParser.SELECT =>
         // Regular select
 
+        // Broadcast hints
+        var withBroadcastedTable = relation
+        if (ctx.hint != null) {
+          val broadcastedTables =
+            hint.mapJoinHint.broadcastedTables.asScala.map(visitTableIdentifier)
+          for (table <- broadcastedTables) {
+            var stop = false
+            withBroadcastedTable = withBroadcastedTable.transformDown {
+              case r @ BroadcastHint(UnresolvedRelation(_, _)) => r
+              case r @ UnresolvedRelation(t, _) if !stop && t == table =>
```

Comment:
I have saw the code of Hive. I think that the first place on tbl_b will be broadcasted.